<launch>
  <arg name="cmd_src" default="teleop"/>
  <arg name="robot" default="nusim"/>
  <arg name="use_rviz" default="true"/>
  <arg name="sensor" default="true"/>

  <node name="tf" pkg="tf2_ros" exec="static_transform_publisher" args="--x 0 --y -0 --z 0 --roll 0 --pitch 0 --yaw 0 --frame-id nusim/world --child-frame-id map"/>
  <node pkg="rviz2" exec="rviz2" args="-d $(find-pkg-share nuslam)/config/just_slam.rviz" if="$(eval '\'$(var use_rviz)\' == \'true\'')"/> 

  <include file="$(find-pkg-share nuturtle_description)/launch/load_one.launch.py" if="$(eval '\'$(var robot)\' == \'nusim\'')" >
    <arg name="use_rviz" value="false"/>
    <arg name="use_jsp" value="true"/>
    <arg name="color" value="green"/>
  </include>
  <include file="$(find-pkg-share nuturtle_control)/launch/start_robot.launch.xml">
    <arg name="cmd_src" value="$(var cmd_src)"/>
    <arg name="robot" value="$(var robot)"/>
    <arg name="use_rviz" value="false"/>
  </include>

  <node pkg="nuslam" exec="slam" name="slam" if="$(eval '\'$(var robot)\' == \'nusim\'')" >
    <param from="$(find-pkg-share nuturtle_description)/config/diff_params.yaml"/>
    <param name="body_id" value="green/base_footprint"/>
    <param name="odom_id" value="green/odom"/>
    <param name="wheel_left" value="wheel_left_joint"/>
    <param name="wheel_right" value="wheel_right_joint"/>
    <param name="sensor" value="$(var sensor)"/>
    <remap from="joint_states" to="blue/joint_states"/>
  </node>

  <node pkg="nuslam" exec="landmarks" name="landmarks" if="$(eval '\'$(var robot)\' == \'nusim\'')" />

</launch>